acceptance/tests/lookup/merge_strategies.rb

```diff
@@ -0,0 +1,194 @@
+test_name 'C99903: merge strategies' do
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+  tmp_environment2 = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath2  = "#{environmentpath}/#{tmp_environment2}"
+
+  master_confdir = master.puppet('master')['confdir']
+  hiera_conf_backup = master.tmpfile(app_type)
+
+  teardown do
+    step "restore default global hiera.yaml" do
+      on(master, "mv #{hiera_conf_backup} #{master_confdir}/hiera.yaml")
+    end
+  end
+
+  step "create global hiera.yaml and environment data" do
+    step "backup global hiera.yaml" do
+      on(master, "cp -a #{master_confdir}/hiera.yaml #{hiera_conf_backup}")
+    end
+
+    create_remote_file(master, "#{master_confdir}/hiera.yaml", <<-HIERA)
+---
+:backends:
+  - yaml
+:yaml:
+  :datadir: "/etc/puppetlabs/code/environments/%{::environment}/hieradata"
+:hierarchy:
+  - "host"
+  - "roles"
+  - "profiles"
+  - "%{::operatingsystem}"
+  - "%{::osfamily}"
+  - "%{::kernel}"
+  - "common"
+:logger: console
+:merge_behavior: deeper
+HIERA
+
+    on(master, "mkdir -p #{fq_tmp_environmentpath}/hieradata/")
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hieradata/host.yaml", <<-YAML)
+---
+profiles:
+  webserver:
+    apache:
+      httpd:
+        modules:
+          - mpm_prefork
+          - php
+          - ssl
+arrayed_hash:
+  - array1:
+      key1: val1
+      key2: val2
+    YAML
+
+    create_remote_file(master, "#{fq_tmp_environmentpath}/hieradata/profiles.yaml", <<-YAML)
+profiles:
+  webserver:
+    apache:
+      httpd:
+        modules:
+          - auth_kerb
+          - authnz_ldap
+          - cgid
+          - php
+          - status
+arrayed_hash:
+  - array1:
+      key1: valB
+      key3: val3
+YAML
+
+    create_sitepp(master, tmp_environment, <<-SITE)
+notify { "hiera_hash: ${hiera_hash ('profiles')['webserver']['apache']['httpd']['modules']}": }
+notify { "lookup1: ${lookup ('profiles')['webserver']['apache']['httpd']['modules']}": }
+notify { "lookup1b: ${lookup ({'name' => 'profiles', 'merge' => 'deep'})['webserver']['apache']['httpd']['modules']}": }
+notify { "hiera_arrayed_hash: ${hiera_array ('arrayed_hash')}": }
+notify { "lookup_arrayed_hash: ${lookup ({'name' => 'arrayed_hash', 'merge' => {'strategy' => 'deep', 'merge_hash_arrays' => true}})}": }
+    SITE
+
+    on(master, "chmod -R 775 #{fq_tmp_environmentpath}")
+    on(master, "chmod -R 775 #{master_confdir}")
```

Comment:
Is changing the permissions on the real `confdir` going to affect/break later tests that run `puppet apply/agent` and assume there are no changes? Do we just need to make the newly generated `hiera.yaml` ownable by `puppet:puppet`?